hail/python/hail/expr/functions.py

```diff
@@ -5134,12 +5137,65 @@ def uniroot(f: Callable, min, max):
         The root of the function `f`.
     """
 
-    new_id = Env.get_uid()
-    lambda_result = to_expr(f(construct_variable(new_id, hl.tfloat64)))
-
-    indices, aggregations = unify_all(lambda_result, min, max)
-    ir = Uniroot(new_id, lambda_result._ir, min._ir, max._ir)
-    return construct_expr(ir, lambda_result._type, indices, aggregations)
+    # Based on:
```

Comment:
Documentation: We should document the new max_iter, epsilon, and tolerance parameters.